autoload/lsp/ui/vim.vim

```diff
@@ -339,34 +347,16 @@ function! s:handle_hover(server, last_req_id, type, data) abort
     endif
 
     if empty(a:data['response']['result'])
-        echom 'No ' . a:type .' found'
+        call s:error_msg('No ' . a:type .' found')
         return
     endif
 
     let l:contents = a:data['response']['result']['contents']
 
-    if type(l:contents) == type('')
```

Comment:
This logic is actually quite important based of the protocol. I tried this PR for [typescript-language-server](https://github.com/theia-ide/typescript-language-server) and doesn't work. Shows empty preview.

https://microsoft.github.io/language-server-protocol/specification#textDocument_hover

Protocol says result can be `null | Hover` where `Hover` can be `MarkedString | MarkedString[] | MarkupContent`. `type MarkedString = string | { language: string; value: string };`

```
export interface MarkupContent {
	/**
	 * The type of the Markup
	 */
	kind: MarkupKind;

	/**
	 * The content itself
	 */
	value: string;
}
```

Out of curiosity for which lang server are you using?

Reply:
ElixirLS

Reply:
@prabirshrestha I have moved that functionality to `lsp#ui#vim#output#preview`.